synapse/storage/databases/main/transactions.py

```diff
@@ -249,7 +250,11 @@ def _set_destination_retry_timings(
                     "retry_interval": retry_interval,
                 },
             )
-        elif retry_interval == 0 or prev_row["retry_interval"] < retry_interval:
+        elif (
+            retry_interval == 0
+            or prev_row["retry_interval"] is None
```

Comment:
I believe this is fixing a separate bug you found?

Reply:
It is